Adjust the design of determining the fields and other aspects (#21)
tc39/proposal-intl-duration-format: README.md

# Overview

* Time Duration is how long something lasts, from the start to end. It can be represented by a single time unit or multiple ones. 

>   + For example, 

Reviewer: Did you change the bullets on purpose? `+` is not a bullet point in markdown, I would suggest that you either revert back to `-` or use `*`.

Author: I used a markdown plugin from VSCode and you can clearly see the bullets in the preview mood.

Reviewer: I misunderstood.

    - 10000 seconds
    - 2 hours 46 minutes 40 seconds

* Every locale has its own way to format duration. 

>   + For example:

Reviewer: Same as above.

Author: ditto.

    - en-US: 1 hour, 46 minutes and 40 seconds
    - fr-FR: 1 heure, 46 minutes et 40 secondes

* There are multiple widths for the time duration.

>   + For example, wide and short

Reviewer: Same as above.

    - 1 hour, 46 minutes and 40 seconds → Wide
    - 1 hr, 46 min, 40 sec → Short

# Motivation

* Users need all types of duration format depending on the requirements of their application. For example, to show how long a flight takes, the duration should be in Short or Narrow format
  + 1 hr 40 min 60 sec → Short 

>   + 1 h 40 m 60 sec  → Narrow

Reviewer: Same as above.

Author: ditto.

# Requirements & Design

In this section, we are going to illustrate each user needs (requirements) and a design for each need (requirement)

## Input Value

>   + Users needs to identify how to input a duration value. For example, if a user needs to format `1000 seconds` , how could the user pass the value to the formatting function.

Reviewer: ```suggestion
  + Users need to identify how to input a duration value. For example, if a user needs to format `1000 seconds` , how could the user pass the value to the formatting function.
```

Author: Done!

### Design

  + Input value will be an object of type `Temporal.Duration` 
  + Example:

     * `const d = new DurationFormat();` 
     * `d.format(new Temporal.Duration.from({hours: 3, minutes: 4});` 

## Formatting width

Users want to determine several types of the formatting width as following

| Format width | Example               |
|--------------|-----------------------|
| Wide         | 1 hour and 50 minutes |
| Short        | 1 hr, 50 min          |
| Narrow       | 1h 50 m               |
| Dotted       | 1: 50: 00             |

### Design

>   + The user can determine the formatting width using a  parameter `style` and the value of this parameter take a ` string` as following:

Reviewer: ```suggestion
  + The user can determine the formatting width using a parameter `style` and the value of this parameter may be one of the following `string`s:
```

Author: Done!

      * `“wide”` 
      * `“short”` 
      * `“narrow”` 
      * `“dotted”` 

## Supported Time Fields

Users need the following fields to be supported

  + Year
  + Month
  + Week
  + Day
  + Hour
  + Minute
  + Second
  + Millisecond
  + Microsecond
  + Nano Second

### Design

* We are going to support the same fields as in [ `Temporal.Duration` ](https://github.com/tc39/proposal-temporal/blob/main/docs/duration.md), which contains:

   * years
   * months

>    * days

Reviewer: weeks

Author: Done

   * Example: 

``` javascript
    fields: [
        'hour',
        'minute',
        'second'
    ]
```

#### NOTE (For stage 3)

* We are discussing to use `largestField` and `smallestField` options to determine the time fields instead of an array.

## Hide zero-value fields

* Users needs to hide zero-value fields depends in some criteria
  + Hide all zero-value fields
    - Hours, Minutes, Seconds (without the `hide` option)
      * 1 hours, 0 minutes and 59 seconds.
    - Hours, Minutes, Seconds (with the `hide` option)
      * 1 hours and 59 seconds.
  + Hide-upper-limit with zero-valued.
    - Hours, Minutes, Seconds, MilliSeconds (without the `hide-upper-limit` option)
      * 0 hr, 20 min, 0 sec, 100 msec .
    - Hours, Minutes, Seconds MilliSeconds(with the `hide-upper-limit` option)
      * 20 min, 0 sec, 100 msec.
  + Hide-lower-limit with zero-valued.

### Design

* Users can set the `hideZeroValues` parameter  by one of the following values:
    - `"all"` // Hide all the fields that have a zero-valued.
    - `"leadingAndTrailing"` // Hide all the zero fields in the leading or the training
    - `"leadingOnly"` 
    - `"trailingOnly"` 

>     - `"none"` // Do not hide any zero-valued fields.

Reviewer: These don't need to be comments.
